sklearn/svm/src/libsvm/svm.cpp

```diff
@@ -2260,7 +2260,7 @@ static void svm_group_classes(const PREFIX(problem) *prob, int *nr_class_ret, in
                 i = j-1;
                 this_label = label[j];
                 this_count = count[j];
-                while(i>=0 && label[i] > this_label)
+                while(i>=0 && label[i] < this_label)
```

Comment:
If we change the order, then y=1 will be the first label, thus assigned to the class `+1`, and y=0 (or -1) will be assigned to the class `-1`.